modules/core/include/opencv2/core/hal/intrin_sse.hpp

```diff
@@ -1032,14 +1032,23 @@ inline _Tpvec operator >= (const _Tpvec& a, const _Tpvec& b) \
 OPENCV_HAL_IMPL_SSE_FLT_CMP_OP(v_float32x4, ps)
 OPENCV_HAL_IMPL_SSE_FLT_CMP_OP(v_float64x2, pd)
 
-#define OPENCV_HAL_IMPL_SSE_64BIT_CMP_OP(_Tpvec, cast) \
+#if CV_SSE4_1
+#define OPENCV_HAL_IMPL_SSE_64BIT_CMP_OP(_Tpvec) \
 inline _Tpvec operator == (const _Tpvec& a, const _Tpvec& b) \
-{ return cast(v_reinterpret_as_f64(a) == v_reinterpret_as_f64(b)); } \
+{ return _Tpvec(_mm_cmpeq_epi64(a.val, b.val)); } \
 inline _Tpvec operator != (const _Tpvec& a, const _Tpvec& b) \
-{ return cast(v_reinterpret_as_f64(a) != v_reinterpret_as_f64(b)); }
+{ return ~(a == b); }
+#else
+#define OPENCV_HAL_IMPL_SSE_64BIT_CMP_OP(_Tpvec) \
+inline _Tpvec operator == (const _Tpvec& a, const _Tpvec& b) \
+{ __m128i cmp = _mm_cmpeq_epi32(a.val, b.val); \
+  return _Tpvec(_mm_or_si128(cmp, _mm_shuffle_epi32(cmp, _MM_SHUFFLE(2, 3, 0, 1)))); } \
```

Comment:
> or

low 32-bits are equal "or" high 32-bits are equal?

Probably `operator !=` should be defined first through "or".

Reply:
Actually if either low or upper are set, the whole 64-bit register needs to be set.  If not, functions like v_check_any/all will not work

Reply:
Not quite sure what you are suggesting in your above comment.

Reply:
Fix `_mm_or_si128` => `_mm_and_si128()` is enough.

Example:
```
v_int64x2 a = v_setall_s64(0x111223344);
v_int64x2 b = v_setall_s64(0x011223344);
```

Reply:
You're right.  It should be AND and not OR.

I'll add some test cases.